src/Console/Command/DescribeCommand.php

```diff
@@ -0,0 +1,227 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+use PhpCsFixer\ConfigurationException\UnallowedFixerConfigurationException;
+use PhpCsFixer\Differ\SebastianBergmannDiffer;
+use PhpCsFixer\FixerDescriptionAwareInterface;
+use PhpCsFixer\FixerFactory;
+use PhpCsFixer\FixerInterface;
+use PhpCsFixer\RuleSet;
+use PhpCsFixer\StdinFileInfo;
+use PhpCsFixer\Tokenizer\Tokens;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Formatter\OutputFormatter;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * @internal
+ */
+final class DescribeCommand extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('describe')
+            ->setDefinition(
+                array(
+                    new InputArgument('name', InputArgument::REQUIRED, 'Name of rule / set.'),
+                )
+            )
+            ->setDescription('Describe rule / ruleset.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $name = $input->getArgument('name');
+
+        if ('@' === substr($name, 0, 1)) {
+            $this->describeSet($input, $output, $name);
+        } else {
+            $this->describeRule($input, $output, $name);
+        }
+    }
+
+    private function describeRule(InputInterface $input, OutputInterface $output, $name)
+    {
+        $fixerFactory = new FixerFactory();
+        $fixers = array();
+
+        foreach ($fixerFactory->registerBuiltInFixers()->getFixers() as $fixer) {
+            $fixers[$fixer->getName()] = $fixer;
+        }
+
+        if (!isset($fixers[$name])) {
+            throw new \InvalidArgumentException(sprintf('Rule "%s" does not exist.', $name));
+        }
+
+        $fixer = $fixers[$name];
+        $description = $fixer instanceof FixerDescriptionAwareInterface ? $fixer->getFixerDescription() : null;
+
+        $output->writeln(sprintf('<info>Description of %s rule.</info>', $name));
+        $output->writeln($fixer->getDescription());
+        $output->writeln('');
+
+        if ($fixer->isRisky()) {
+            $output->writeln('<error>Fixer applying this rule is risky.</error>');
+
+            if ($description && $description->getRiskyDescription()) {
+                $output->writeln($description->getRiskyDescription());
+            }
+
+            $output->writeln('');
+        }
+
+        if ($this->isFixerConfigurable($fixer)) {
+            $output->writeln('<comment>Fixer is configurable.</comment>');
+
+            if ($description && $description->getConfigurationDescription()) {
+                $output->writeln($description->getConfigurationDescription());
```

Comment:
You presented a whole Fixer description info. It's already there - `FixerDescription` object.
Let us not work with anonymous array structures.
Instead of array I work with DTO object. 

Then:
- `short` is already on fixer interface, no need for duplication
- `examples/out` should be autogenerated by the fixer, so we won't have an issue when one will provide in `out` sth not matching real Fixer behaviour
- `order` - let us skip it for now

All the other data is already there.

---------

The issue is how to represent string description of possible configuration. So your `configuration/description` key.